### PR TITLE
Weeder linting

### PR DESCRIPTION
**Pull request by bergmark**
The libs are now weed free, things to do:

* Set up travis job
* Fix remaining weeds in test suite

### CODE REVIEW

File: tests/Types.hs [hunk 1]
```diff
@@ -125,11 +125,11 @@ failure func msg v = counterexample
 newtype BCEDay = BCEDay Day
   deriving (Eq, Show)
 
-zeroDay :: Day
-zeroDay = fromGregorian 0 0 0
-
 instance Arbitrary BCEDay where
     arbitrary = fmap (BCEDay . ModifiedJulianDay . (+ toModifiedJulianDay zeroDay)) arbitrary
+        where
+            zeroDay :: Day
```
Comment (phadej) [inline tests/Types.hs]: weird indentation, is it intentional?

File: tests/DataFamilies/Types.hs [hunk 2]
```diff
@@ -18,7 +18,7 @@ import Types (ApproxEq(..))
 
 data family Nullary a
 data instance Nullary Int  = C1 | C2 | C3 deriving (Eq, Show)
-data instance Nullary Char = C4           deriving (Eq, Show)
+-- data instance Nullary Char = C4           deriving (Eq, Show)
```
Comment (phadej) [inline tests/DataFamilies/Types.hs]: I wonder if this was used to trigger / test some bug/regression. Hopefully not.

File: tests/Types.hs (hunk 1 above)
Comment (bergmark) [inline tests/Types.hs]: Intentional :-) how would you do it?

Comment (phadej) [inline tests/Types.hs]:
```haskell
instance
    arbitrary
      where
        zeroDay
```

I.e. use half-indentation for `where`

**Comment by bergmark** [comment]
Probably better to start from scratch if we want to include this.

**Pull request closed without merging**